src/erlcloud_firehose_impl.erl

```diff
@@ -0,0 +1,177 @@
+%% -*- mode: erlang;erlang-indent-level: 4;indent-tabs-mode: nil -*-
```

Comment:
from _impl module perspective I see no value in having a separate one. 
To have a better picture is would be nice to see full firehose thing as it might be that separating out _impl module is unnecessary at all and be put directly into `erlcloud_firehose`